[refactor] 일정 삭제 시 해당 월의 검색범위에서 일정 찾아 삭제

clara-shin opened the pull request:
# 🚀 풀 리퀘스트 제안
closes #142 
<!-- 이슈 지울때 작성해 주세요. -->
<!-- closes #issue-number -->
<!-- 이슈 여러 개 일 경우, closes #1, closes #2 -->
<!-- https://docs.github.com/ko/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

## 📋 작업 내용

- Firestore는 복합 쿼리에 제한이 있어 startDate와 endDate를 동시에 범위 쿼리로 사용할 수 없음
- Firestore는 복합 쿼리(여러 필드를 기반으로 하는 쿼리)를 수행할 때 특정 인덱스가 필요, 생성함
<img width="935" alt="스크린샷 2024-08-07 오전 1 42 44" src="https://github.com/user-attachments/assets/5de0bb3d-c956-4308-9ad6-0cd367dc2d8f">

- scheduleSlice.ts 파일에,
    - handleConfirmDelete 함수에서 삭제할 일정의 startDate와 endDate 안에서(지정된 기간 내에서) 해당 userNo를 가진 일정만 삭제되도록 수정
    - 혹시 모를 userNo를 가진 다른 일정이 삭제되는 것을 방지
    - Firestore의 복합 인덱스 요구사항 충족

- handleConfirmDelete 함수를 useScheduleDelete 훅으로 분리 함

## 🔧 변경 사항

- [ ] 📃 README.md
- [ ] 📦 package.json
- [ ] 🔥 파일 삭제
- [ ] 🧹 그 외 ex) .gitignore 등

주요 변경 사항을 요약해 주세요.

## 📸 스크린샷 (선택 사항)

수정된 화면 또는 기능을 시연할 수 있는 스크린샷을 첨부해 주세요.

## 📄 기타

추가적으로 전달하고 싶은 내용이나 특별한 요구 사항이 있으면 작성해 주세요.

clara-shin committed:
fix: 파일명 소문자로 변경

clara-shin committed:
refactor: 일정 삭제 시 해당 월의 범위를 벗어나는 일정이 삭제되지 않도록 수정 (#142)

- Firestore는 복합 쿼리에 제한이 있어 startDate와 endDate를 동시에 범위 쿼리로 사용할 수 없음
- Firestore는 복합 쿼리(여러 필드를 기반으로 하는 쿼리)를 수행할 때 특정 인덱스가 필요
    => Firestore에서 인덱스 생성했음

- scheduleSlice.ts 파일에,
    - handleConfirmDelete 함수에서 삭제할 일정의 startDate와 endDate 안에서(지정된 기간 내에서)
    - 해당 userNo를 가진 일정만 삭제되도록 수정
    - 혹시 모를 userNo를 가진 다른 일정이 삭제되는 것을 방지
    - Firestore의 복합 인덱스 요구사항 충족

clara-shin committed:
refactor: handleConfirmDelete 함수 분리 (#142)

- handleConfirmDelete 함수를 useScheduleDelete 훅으로 분리

clara-shin committed:
refactor: React.strictMode 적용, console.log 제거 (#142)

clara-shin committed:
feat: 일정 삭제 시 토스트 메시지 추가 (#142)

kimisadev27 commented on src/components/dailySchedule/DailySchedule.tsx:
```diff
@@ -20,15 +20,18 @@ const DailySchedule = ({ date, schedules }: DailyScheduleProps) => {
   // 컴포넌트가 마운트 되거나 date, schedules가 변경될 때 필터링 실행
   useEffect(() => {
     filterSchedules();
+    console.log(filterSchedules());
```
console.log 삭제 부탁드려요!

clara-shin replied:
삭제완료!

kimisadev27 commented on src/hooks/useScheduleDelete.ts:
```diff
@@ -0,0 +1,55 @@
+
+      // navigate해주기 전에 삭제되었다고 토스트ui 띄우기 코드 필요
+      toastTrigger('일정이 삭제되었습니다.');
+      navigate('/schedule');
```
PATH.SCHEDULE 사용하는걸 권장드립니다!

clara-shin replied:
변경완료!

kimisadev27 commented on src/pages/DailyScheduleDetail.tsx:
```diff
@@ -89,7 +69,7 @@ const DailyScheduleDetail = () => {
           isOpen={isModalOpen}
           onClose={handleCancelDelete}
-          onConfirm={() => handleConfirmDelete(schedule.userNo)}
+          onConfirm={() => handleConfirmDelete()}
```
매개변수가 안들어가면
onConfirm={handleConfirmDelete}  로 하셔도 됩니다!

clara-shin replied:
네 변경했습니다!👍

clara-shin committed:
refactor: navigate 경로 변경 (#142)

PATH 상수값으로 변경
:id 를 문자열로 바꿔서 전달

clara-shin committed:
remove: useScheduleDelete.ts 파일 삭제 후 DailyScheduleDetail에 병합

clara-shin committed:
feat: 일정 삭제 및 조회 api 연동 (#142)

- DailySchedule에서 받아온 id값을 통해 해당 일정을 가져오는 api 연동
    - 현재는 사용하지 않음
    - 현재는 DailyScheduleDetail에서 useFetchScheduleDetail을 사용하여 일정을 가져오고 있음
- DailySchedule에서 조회 된 일정의 id값과 그 일정의 startDate, endDate를 통해 해당 일정을 삭제하는 api 연동
    - 현재 사용 중

clara-shin committed:
refactor: DailyScheduleDetail에서 하루일정 조회

- 이미 useFetchSchedule로 가져온 schedules를 사용자
    - 다시 데이터를 호출하지 않도록 함

clara-shin committed:
feat: id로 일정 삭제 기능 추가 (#142)

- id, startDate, endDate로 db에서 찾아 삭제하는 기능 추가
- 삭제 후 토스트 메시지 출력

clara-shin committed:
Merge branch 'develop' into refactor/func-handleConfirmDelete-#142
